### PR TITLE
chore: remove typescript legacy support option

### LINKED ISSUE #439: [Bug]: fix type errors revealed by Typescript 5.0
### Package name

core

### Package version

9.0.0-next

### Reproduction steps

- Make sure Typescript 5 is used (https://github.com/AmadeusITGroup/otter/pull/402 merged)
- remove `    "ignoreDeprecations": "5.0",` from `tsconfig.base.ts`
- install and build

### Current result

- many type issues appear at build time

### Expected result

- type issues should be all fixed

### Additional comments

_No response_

### PR DESCRIPTION
**Pull request by kpanot**
## Related issues

- :bug: Fixes #439

### CODE REVIEW

File: packages/@ama-sdk/core/src/utils/json-token.ts [hunk 1]
```diff
@@ -148,8 +148,8 @@ export function createJweEncoder(aesTagLengthInBits = 128, useHeaderAsAAD = fals
   const base64Encoder = createBase64UrlEncoder();
   const stringEncoder = typeof window.TextEncoder !== 'undefined' ? new TextEncoder() : new Encoder();
 
-  return async ({publicKey, keyId}: {publicKey: Promise<CryptoKey> | CryptoKey; keyId: string}, jwePayload: Record<string, unknown>, publicProperties: string[]) => {
-    const jweHeader = {
+  return async ({ publicKey, keyId }: { publicKey: Promise<CryptoKey> | CryptoKey; keyId: string }, jwePayload: Record<string, unknown>, publicProperties: string[]) => {
+    const jweHeader: Record<string, unknown> = {
```
Comment (vscaiceanu-1a) [inline packages/@ama-sdk/core/src/utils/json-token.ts:152]: why `unknown`?

File: packages/@o3r/analytics/plugins/webpack/build-metrics/build-stats.plugin.ts [hunk 2]
```diff
@@ -44,9 +44,9 @@ export class BuildStatsPlugin implements WebpackPluginInstance {
       .forEach(([hookName, hook]) => hook.intercept(this.makeInterceptorFor('Compiler')(hookName)));
 
     Object.entries(compiler.resolverFactory.hooks)
-      .filter(([hookName]) => !!compiler.resolverFactory.hooks[hookName])
-      .map(([hookName]) => [hookName, compiler.resolverFactory.hooks[hookName]])
-      .forEach(([hookName, hook]) => hook.intercept(this.makeInterceptorFor('Compiler')(hookName)));
+      .filter(([hookName]) => !!compiler.resolverFactory.hooks[hookName as 'resolver' | 'resolveOptions'])
```
Comment (vscaiceanu-1a) [inline packages/@o3r/analytics/plugins/webpack/build-metrics/build-stats.plugin.ts:47]: `'resolver' | 'resolveOptions'` can be extracted in a type

**Review by vscaiceanu-1a — COMMENTED** [review]
I think we can reduce the number of `as any`

File: packages/@ama-sdk/core/src/utils/json-token.ts (hunk 1 above)
Comment (kpanot) [reply packages/@ama-sdk/core/src/utils/json-token.ts:152]: Coming from `jwePayload` type, I kept the original type

**Review by matthieu-crouzet — APPROVED** [review]

**Review by fpaul-1A — APPROVED** [review]

**Pull request merged**